src/strings/ops.c

```diff
@@ -1693,13 +1714,51 @@ MVMString * MVM_string_join(MVMThreadContext *tc, MVMString *separator, MVMObjec
     }
     result->body.num_graphs = total_graphs;
 
-    /* If we just collect all the things as strands, are we within bounds, and
-     * will be come out ahead? */
-    if (total_strands < MVM_STRING_MAX_STRANDS && 16 <= total_graphs / total_strands) {
-        /* XXX TODO: Implement this, conditionalize branch thing below. */
+    /* If the separator and pieces are all strands, and there are
+     * enough strands of a large enough size, just copy them directly. */
```

Comment:
Can you change it to this please:
```
/* If the separator and pieces are all strands, and
 * there are on average at least 16 graphemes in each of the strands */
if (all_strands && total_strands < MVM_STRIND_MAX_STRANDS
           && total_strands * 16 <= total_graphs)
```
The conditional becomes much clearer and clearer comment.